src/main/kotlin/xyz/atrius/waystones/data/config/Localization.kt

```diff
@@ -0,0 +1,42 @@
+package xyz.atrius.waystones.data.config
+
+import org.bukkit.configuration.file.YamlConfiguration
+import xyz.atrius.waystones.utility.KotlinPlugin
+import xyz.atrius.waystones.utility.translateColors
+import java.io.File
+import java.io.InputStreamReader
+import java.text.MessageFormat
+import java.util.*
+
+class Localization(val plugin: KotlinPlugin, localeTag: String) {
+    private var configFile = File(plugin.dataFolder, "locale-$localeTag.yml")
+    private val config = YamlConfiguration()
+    private var locale: Locale = Locale.getDefault()
+
+    init {
+        reload(localeTag)
+    }
+
+    fun localize(key: String, vararg args: Any?): String {
+        return MessageFormat(config.getString(key)!!, locale)
+                .format(*args.map { it ?: "null" }.toTypedArray()).translateColors()
+    }
+
+    fun reload(localeTag: String) {
+        configFile = File(plugin.dataFolder, "locale-$localeTag.yml")
+
+        if (!configFile.exists()) {
+            plugin.saveResource(configFile.name, false)
+            config.load(configFile)
+        } else {
+            val defaultConfigFile = plugin.getResource(configFile.name) ?: plugin.getResource("locale-en.yml")!!
```

Comment:
Please avoid use of the `!!` operator as mentioned above.

Reply:
If there is no `locale-en.yml` in the jar, an Exception should probably be thrown?

Reply:
If we should throw an exception I think it would be best to ensure its clear in its meaning by throwing our own instead of relying on a `NullPointerException`.